### PR TITLE
Removed deprecated from_df method from ColumnDataSource

### PR DESCRIPTION
**Pull request by canavandl**
ref issue: #2581 

This is removes a deprecated ColumnDataSource method "from_df". It has been supplanted by the __init__ constructor.

### CODE REVIEW

File: examples/glyphs/data_tables_server.py [hunk 1]
```diff
@@ -132,7 +132,7 @@ def update_data(self):
             df = df[df["drv"] == self.drive_filter]
         if self.class_filter:
             df = df[df["class"] == self.class_filter]
-        self.source.data = ColumnDataSource.from_df(df)
+        self.source.data = ColumnDataSource._data_from_df(df)
```
Comment (bryevdv) [inline examples/glyphs/data_tables_server.py]: This is problematic. Our examples should never rely on private APIs.

Comment (canavandl) [inline examples/glyphs/data_tables_server.py]: you're right. the fix is quick

**Comment by canavandl** [comment]
This method also appears to be used in odo:

https://github.com/blaze/odo/blob/master/odo/backends/bokeh.py#L18

**Comment by canavandl** [comment]
I agree. Implementing on some AJAXDataSource stuff, I realized I wanted access to the df parsing/validation machinery in order to create appropriate JSON to return as a http response.

I think the method rename (from_df -> parse_df) and implementation tweaks (classmethod -> staticmethod, and other little tweaks) gets us there.

File: bokeh/models/sources.py [hunk 2]
```diff
@@ -138,23 +137,6 @@ def _data_from_df(df):
...
-
-        Returns:
-            dict(str, list)
-
-        """
-        import warnings
-        warnings.warn("Method deprecated in Bokeh 0.9.3")
-        return cls._data_from_df(data)
-
     def to_df(self):
```
Comment (bryevdv) [inline bokeh/models/sources.py]: Can we check if this is actually used anywhere at all? If not, I would suggest perhaps removing it.

Comment (canavandl) [inline bokeh/models/sources.py]: I don't entirely understand what you're referring to removing or not.

I did remove the from_df method entirely (made it a private method) in an earlier commit, but we decided that exposing the method was likely useful. We use it in a bokeh server example, odo uses it to support bokeh as a backend and I've used it to validate data to send as JSON to an AJAXDataSource (and think it'll be more common going forward).

The original naming and implementation of the method wasn't very good, so this commit renames it to "parse_df" (not a great name, but at least there is a helpful docstring now) and reimplements it.

Comment (bryevdv) [inline bokeh/models/sources.py]: Sorry I was talking about to_df I don't remember what if anything uses it.

Comment (canavandl) [inline bokeh/models/sources.py]: It apparently gets used in charts/_attributes.py:
https://github.com/bokeh/bokeh/blob/master/bokeh/charts/_attributes.py#L112
and in crossfilter/models.py and plugins.py

I know odo also uses in to support bokeh as a backend too (ColumnDataSource -> df)

Whether any actual users use to_df, I doubt it?

Comment (bryevdv) [inline bokeh/models/sources.py]: Ah ok, let's just leave it for now.

**Comment by bryevdv** [comment]
So, I'd actually suggest  we re-think this a little. I'd like to investigate if there are any problems/downsides to letting `.data` be *either* a `dict(string, seq)` or just stay a Dataframe, if that is what is passed in. i.e., letting `ColumnDataSource` be a thin wrapper around anything that looks like a "dict of arrays".

@bokeh/dev thoughts?

**Comment by bryevdv** [comment]
I now think we should wait on this. There may be some changes to data sources in general coming. I am going to close this for now

**Pull request closed without merging**